detox/e2e/test/autocomplete/edit_channel.e2e.js

```diff
@@ -11,6 +11,10 @@ describe('Autocomplete', () => {
         await toChannelScreen(user);
     });
 
+    afterAll(async () => {
+        await logoutUser();
+    });
```

Comment:
I'm curious about what was your observation here. Were there data/user info remnants that were not being cleared before the test start? I've tested several times with and without all of `afterAll` hooks and I'm getting the same results.
At [init.js](https://github.com/mattermost/mattermost-mobile/blob/master/detox/e2e/init.js#L12), we're launching the app before each test file which is said to launch the app cleanly (remove > install > open). I wonder if there's something to change there instead of dealing with the individual `afterAll` hook.

Reply:
Although what you say is true, as a dev I want to run my tests with a continuous workflow, so I can use something like `detox --reuse` with this option, the app is not uninstalled / installed. That way tests run as soon as I make changes and there is no need to wait. So by having the tests log out the user, we ensure that all tests pass.

Reply:
Thanks and well noted on dev workflow. However, as we grow the number of test files, I don't see dev/contributors to run all of the tests in their local machine. They should be running specific test only, like:
```
npm run e2e:ios-test -- e2e/test/channel_info/header.e2e.js
```

Here are my thoughts in using `afterAll`:
- It works well in happy path only - all tests are executed and passed.
- Dev/contributor needs to keep track of test state and put the state of the last test so that `afterAll` will be able to continue the flow.
- It will be a challenge to run test in isolation, like when someone wants to run `it.only` for test file with multiple tests.
- It is not guaranteed to run, especially when something breaks/fails in the middle of the test.

But I'm good as is. Let's observe and improve if we'll actually encounter the points above in the future.

Reply:
very good points.. I'm ok if you want to revert that. One thing that I'm 4/5 is that we should add how to run selected or if a singe test file in the docs

Reply:
I'm fine as is and I'd like to observe it first before making a change in the future. Sure, I'll send docs PR about it.